src/Repository/UserRepository.php

```diff
@@ -115,4 +116,18 @@ public function getPaginatedUserList(
 
         return $query;
     }
+
+    /**
+     * @param User $user
+     * @param Image $avatar
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function saveAvatar(User $user, Image $avatar): void
```

Comment:
Why another method? Instead of `$this->userRepository->saveAvatar($user, $image);` you can use:
```
$user->setAvatar($image);
$this->userRepository->save($user);
```